oauth-server/src/main/java/io/strimzi/kafka/oauth/server/OAuthKafkaPrincipal.java

```diff
@@ -35,6 +37,19 @@ public BearerTokenWithPayload getJwt() {
         return jwt;
     }
 
+    public static OAuthKafkaPrincipal popCurrentPrincipal() {
+        OAuthKafkaPrincipal current = currentTL.get();
+        currentTL.remove();
+        return current;
+    }
+
+    public static void pushCurrentPrincipal(OAuthKafkaPrincipal principal) {
+        if (currentTL.get() != null) {
+            throw new RuntimeException("Internal error - current principal already set");
```

Comment:
If these don't support stack-like push and pop semantics then are the method names really the right ones?

Reply:
Fair point. Will rename.